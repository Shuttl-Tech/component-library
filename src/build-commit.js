// Utility to make a build commit with a version number

const fs = require('fs');
const { spawnSync } = require('child_process');

const { version } = JSON.parse(fs.readFileSync('./package.json', { encoding: 'utf-8' }));

const options = {
	stdio: 'inherit'
};

spawnSync('git', ['add', '.'], options);
spawnSync('git', ['commit', '-m', `Build version ${version}.`, '--no-verify'], options);
