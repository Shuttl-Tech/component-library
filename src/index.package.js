export { Box } from 'components/Box';
export { CollapsibleBox } from 'components/CollapsibleBox';
export { ModularBox } from 'components/ModularBox';
export { SplitEvenBox } from 'components/SplitEvenBox';
export { GenericPill } from 'components/GenericPill';
export { InPlaceTooltip } from 'components/InPlaceTooltip';

export { default } from 'components/IncorrectUsageComponent';
