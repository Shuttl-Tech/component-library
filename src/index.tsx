export { default as Button } from 'components/Button';
export { default as FormProgressBar } from 'components/FormProgressBar';
