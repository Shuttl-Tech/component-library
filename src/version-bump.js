// Utility to upgrade package.json version

const fs = require('fs');
const { titleCase } = require('voca');
require('colors');

const { box } = require('./utils/cli');

const UPGRADE_MODE = { PATCH: 'patch',
	MINOR: 'minor',
	MAJOR: 'major'
};

const UPGRADE_MODES = [UPGRADE_MODE.PATCH, UPGRADE_MODE.MINOR, UPGRADE_MODE.MAJOR];

const UPGRADE = process.env.UPGRADE && (UPGRADE_MODES.includes(process.env.UPGRADE.toLowerCase()))? process.env.UPGRADE : UPGRADE_MODE.PATCH;

let file = JSON.parse(fs.readFileSync('./package.json', { encoding: 'utf-8' }));
let oldVersion = file.version;
let [major, minor, patch] = oldVersion.split('.').map(n => parseInt(n, 10));

switch (UPGRADE) {
	case UPGRADE_MODE.MINOR: minor++; break;
	case UPGRADE_MODE.MAJOR: major++; break;
	default: /* PATCH */ patch++;
}

let version = [major, minor, patch].join('.');
file = JSON.stringify({ ...file, version }, null, '\t') + '\n';

fs.writeFileSync('./package.json', file);

let message = `⏫ ${titleCase(UPGRADE)} ⏫ \n🎉 upgrade from version 🎉\n${oldVersion} 👉 ${version} 🎉 `;

box({ message });

