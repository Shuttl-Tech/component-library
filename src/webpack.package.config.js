// Configuration for library package generation

const path = require('path');

module.exports = {
	entry: './src/index.package.js',
	output: {
		path: __dirname + '/',
		publicPath: '/',
		filename: 'index.package.generated.js',
		libraryTarget: 'umd'
	},
	module: {
		rules: [
			{
				test: /\.jsx?$/,
				exclude: /node_modules/,
				use: {
					loader: 'babel-loader',
					options: {
						presets: ['env']
					}
				}
			},
			{
				test: /\.css$/,
				loader: 'style-loader!css-loader?modules&importLoaders=1&localIdentName=[sha512:hash:base32:5]-[local]'
			}
		]
	},
	resolve: {
		extensions: ['*', '.js', '.jsx'],
		alias: {
			'react': path.resolve(__dirname, './node_modules/react'),
			'react-dom': path.resolve(__dirname, './node_modules/react-dom'),
		},
		modules: [
			'src',
			'node_modules'
		]
	},
	externals: {
		'react': 'react',
		'react-dom': 'react-dom'
	}
};
