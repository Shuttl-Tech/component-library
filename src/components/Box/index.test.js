import React from 'react';
import renderer from 'react-test-renderer';

import { Box } from '../../../../src/index';

it('Box tree looks like how it should', () => {
	let props = {
		match: {
			params: {
				id: 1
			}
		}
	};

	const component = renderer.create(<Box {...props} />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
