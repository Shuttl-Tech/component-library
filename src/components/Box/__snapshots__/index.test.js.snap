// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box tree looks like how it should 1`] = `
<div
  className="box component--box"
/>
`;
